Added RedHunt Labs API

owais-redhunt opened the pull request:
RedHunt Labs' Attack Surface Recon APIs are designed to provide comprehensive domain intelligence and reconnaissance capabilities, with a free community plan.

By leveraging our extensive in-house database, which contains over 6 billion records, including domains, subdomains, third-party SaaS, data leaks, and intelligent correlations, our API empowers you to enhance your Attack Surface Management and InfoSec workflows.

m8sec commented:
Hi @owais-redhunt,

This is great and really appreciate the PR! 

I created a free account for testing and was getting stuck in the `while 1` loop. Even when no results were found the loop continued until I eventually received `403: limit has been reached`.

The code below for `redhuntlabs.py` *should* solve this issue but I was not able to test. If you can confirm and resubmit that would be awesome!

```python
     def run(self):
        # API key check
        if not self.config.redhuntlabs['X-BLOBR-KEY']:
            logging.debug(f'Skipping {self.name}, API key(s) not found')
            return False

        page = 1
        url = 'https://reconapi.redhuntlabs.com/community/v1/domains/subdomains?domain={}&page_size=10&page={}'
        headers = {"X-BLOBR-KEY": self.config.redhuntlabs['X-BLOBR-KEY']}

        while page:
            try:
                resp = web_request(url.format(self.domain, page), headers=headers, timeout=self.args.timeout)
                status_code = get_statuscode(resp)

                if status_code == 200:
                    if resp.json()['subdomains']:
                        for sub in resp.json()['subdomains']:
                            self.report_q.add({'Name': f'{sub}.{self.domain}', 'Source': self.name})
                        page += 1
                    else:
                        logging.debug(f'{self.name}: No results found (Pg. {page}) closing thread')
                        page = False
                elif status_code == 403:
                    raise Exception(f'API Limit reached ({status_code})')
                else:
                    raise Exception(f'Web request failed to {url} ({status_code})')

            except Exception as e:
                logging.debug(f'{self.name.upper()} ERR: {e}')
                page = False
```

owais-redhunt commented:
Hey there, thanks for your kind words.

I fixed the issue like you said, I don't seem to have the problem anymore. Test it and let me know. Much love to projects like these, they're the unsung heroes of the world.

P.S.: Saw a type in chaos and fixed that too.

m8sec commented on subscraper/modules/redhuntlabs.py:
```diff
@@ -24,24 +23,25 @@ def run(self):
-        headers = {"X-BLOBR-KEY": self.config.redhuntlabs['X-BLOBR-KEY']}
-
         page = 1
+        url = f'https://reconapi.redhuntlabs.com/community/v1/domains/subdomains?domain={self.domain}&page_size=10&page={}'
```
The domain is added in the URL on line 32, with the correct page variable. please see my code in the comments

m8sec reviewed (approved):
Good catch on the chaos typo 🤦‍♂️
